File: frontend/src/app/controller/service/admin/coupon/InfluencerAdmin.service.ts
import { Injectable } from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {BehaviorSubject, Observable} from 'rxjs';

import {environment} from 'src/environments/environment';

import {InfluencerDto} from 'src/app/controller/model/coupon/Influencer.model';
import {InfluencerCriteria} from 'src/app/controller/criteria/coupon/InfluencerCriteria.model';
import {AbstractService} from 'src/app/zynerator/service/AbstractService';


@Injectable({
  providedIn: 'root'
})
export class InfluencerAdminService extends AbstractService<InfluencerDto, InfluencerCriteria> {
     constructor(private http: HttpClient) {
        super();
        this.setHttp(http);
        this.setApi(environment.apiUrl + 'admin/influencer/');
    }

    public constrcutDto(): InfluencerDto {
        return new InfluencerDto();
    }

    public constrcutCriteria(): InfluencerCriteria {
        return new InfluencerCriteria();
    }
}

File: backend-ms1/src/main/java/ma/zs/zyn/service/facade/admin/coupon/InfluencerAdminService.java
package ma.zs.zyn.service.facade.admin.coupon;

import java.util.List;
import ma.zs.zyn.bean.core.coupon.Influencer;
import ma.zs.zyn.dao.criteria.core.coupon.InfluencerCriteria;
import ma.zs.zyn.zynerator.service.IService;



public interface InfluencerAdminService extends  IService<Influencer,InfluencerCriteria>  {
    Influencer findByUsername(String username);
    boolean changePassword(String username, String newPassword);





}
